src/zk.py

```diff
@@ -546,28 +554,28 @@ def release_if_hold(self, lock_type, wait=0, read_lock=False):
             holders = self.get_lock_contenders(lock_type, read_lock=read_lock)
         else:
             holders = [self.get_current_lock_holder(lock_type)]
-        if helpers.get_hostname() not in holders:
+        if self.get_lock_contender_name() not in holders:
             return True
         return self.release_lock(lock_type, wait)
 
     def get_host_alive_lock_path(self, hostname=None):
-        return _get_host_path(self.HOST_ALIVE_LOCK_PATH, hostname)
+        return self._get_host_path(self.HOST_ALIVE_LOCK_PATH, hostname)
 
     def get_host_maintenance_path(self, hostname=None):
-        return _get_host_path(self.HOST_MAINTENANCE_PATH, hostname)
+        return self._get_host_path(self.HOST_MAINTENANCE_PATH, hostname)
 
     def get_host_quorum_path(self, hostname=None):
-        return _get_host_path(self.QUORUM_MEMBER_LOCK_PATH, hostname)
+        return self._get_host_path(self.QUORUM_MEMBER_LOCK_PATH, hostname)
 
     def get_host_prio_path(self, hostname=None):
-        return _get_host_path(self.HOST_PRIO_PATH, hostname)
+        return self._get_host_path(self.HOST_PRIO_PATH, hostname)
 
     def get_simple_primary_switch_try_path(self, hostname=None):
-        return _get_host_path(self.SIMPLE_PRIMARY_SWITCH_TRY_PATH, hostname)
+        return self._get_host_path(self.SIMPLE_PRIMARY_SWITCH_TRY_PATH, hostname)
 
     def get_election_vote_path(self, hostname=None):
         if hostname is None:
-            hostname = helpers.get_hostname()
+            hostname = self.get_lock_contender_name()
```

Comment:
I'm not sure if this is the right place to move from plain hostname to lock_contender_name.